packages/admin/src/pages/super-admin-dashboard/manage-departments/index.page.tsx

```diff
@@ -0,0 +1,96 @@
+import {
+  Button,
+  QuestionPageGetServerSideProps,
+  SummaryList,
+} from 'gap-web-ui';
+import { getAllDepartments } from '../../../services/SuperAdminService';
+import { getUserTokenFromCookies } from '../../../utils/session';
+import InferProps from '../../../types/InferProps';
+import getConfig from 'next/config';
+import Meta from '../../../components/layout/Meta';
+import CustomLink from '../../../components/custom-link/CustomLink';
+import { GetServerSidePropsContext } from 'next';
+import { Department } from '../types';
+import { Row } from 'gap-web-ui/dist/cjs/components/summary-list/SummaryList';
+
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+  const fetchPageData = async (jwt: string) => ({
+    departments: await getAllDepartments(jwt),
+    userId: context.query?.userId || '',
+  });
+
+  const res = await QuestionPageGetServerSideProps({
+    context,
+    fetchPageData,
+    jwt: getUserTokenFromCookies(context.req),
+    onErrorMessage: 'Failed to load departments, please try again later.',
+  });
+
+  return res;
```

Comment:
Misusing this function a bit - intended for question pages only. If this doesn't have a form, then its not worth using this

Reply:
Fixed that